core/src/com/unciv/logic/city/IConstruction.kt

```diff
@@ -84,13 +84,24 @@ interface INonPerpetualConstruction : IConstruction, INamed, IHasUniques {
 
 
 class RejectionReasons: HashSet<RejectionReasonInstance>() {
-    
+
     fun add(rejectionReason: RejectionReason) = add(RejectionReasonInstance(rejectionReason))
-    
+
     fun contains(rejectionReason: RejectionReason) = any { it.rejectionReason == rejectionReason }
 
-    fun filterTechPolicyEraWonderRequirements(): List<RejectionReasonInstance> {
-        return filterNot { it.rejectionReason in techPolicyEraWonderRequirements }
+    fun isOKIgnoringRequirements(
+        ignoreTechPolicyEraWonderRequirements: Boolean = false,
+        ignoreResources: Boolean = false
+    ): Boolean {
+        if (!ignoreTechPolicyEraWonderRequirements && !ignoreResources) return isEmpty()
+        if (!ignoreTechPolicyEraWonderRequirements)
+            return none { it.rejectionReason != RejectionReason.ConsumesResources }
+        if (!ignoreResources)
+            return none { it.rejectionReason !in techPolicyEraWonderRequirements }
+        return none {
```

Comment:
Doesn't code for mean well never get here?
We check if notA notB
Then we check notA
Then notB
In each check we have a return
So we should never be able to get to this line

Reply:
Sorry my bad

Reply:
Actually I though you'd pick up on this getting called with only a limited subset of 4 bool[2] combos - one branch is indeed unused, the false false one if I remember correctly - _no the other one_ - which is why all 4 should be there even with 1 unused. :zany_face: